File: dsa using java/SelectionSort.java
import java.io.*;

class SelectionSort
{
	public static void main(String args[])
	{
		try
		{
			BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
			
			System.out.print("Enter the number of elements that you want to sort : ");
			int n=Integer.parseInt(br.readLine());
			int arr[]=new int[n];
			
			for(int i=0;i<n;i++)
			{
				System.out.print("Enter the element # "+(i+1)+" : ");
				arr[i]=Integer.parseInt(br.readLine());
			}
			
			Selection_Sort(arr);
			System.out.print("Your Sorted array is :\n");
			for(int i=0;i<arr.length;i++)
				System.out.print(arr[i]+" , ");
		}
		catch(Exception e)
		{
			System.out.println(e);
		}
	}
	
	public static void Selection_Sort(int []arr)
	{
		for(int i=0;i<arr.length-1;i++)
		{
			int minIndex=i;
			for(int j=i+1;j<arr.length;j++)
				if(arr[minIndex]>arr[j])
					minIndex=j;
			int temp=arr[minIndex];
			arr[minIndex]=arr[i];
			arr[i]=temp;
		}
	}
}

File: dsa new/Sorting/Selection Sort.c
#include<stdio.h>
void selectionSort(int arr[],int n)
{
    int minIndex,i,j,temp;
    for(i=0;i<n-1;i++)
    {
        minIndex=i;
        for(j=i+1;j<n;j++)
            if(arr[j]<arr[minIndex])
                minIndex=j;
        temp=arr[i];
        arr[i]=arr[minIndex];
        arr[minIndex]=temp;
    }
}

void selectionSortIndexed(int arr[],int a,int b)
{
    int i,j,temp,minIndex;
    for(i=a;i<b;i++)
    {
        minIndex=i;
        for(j=i+1;j<=b;j++)
            if(arr[j]<arr[minIndex])
                minIndex=j;
        temp=arr[i];
        arr[i]=arr[minIndex];
        arr[minIndex]=temp;
    }
}
int main()
{
    int arr[]={3,8,1,23,7,9,24,6,12};
    int n=9,i;
    selectionSortIndexed(arr,3,6);
    for(i=0;i<n;i++)
        printf("%d\n",arr[i]);
}

